tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/PublicAssetMethodContract.scala

```diff
@@ -0,0 +1,628 @@
+import org.apache.james.utils.DataProbeImpl
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.{BeforeEach, Test}
+import play.api.libs.json.Json
+
+import scala.jdk.CollectionConverters._
+
+object PublicAssetMethodContract {
+  case class UploadResponse(blobId: String, contentType: String, size: Long)
+}
+
+trait PublicAssetMethodContract {
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString)
+      .addUser(BOB.asString(), BOB_PASSWORD)
+      .addUser(ANDRE.asString(), ANDRE_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .addHeader(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .build()
+      .log().ifValidationFails()
+  }
+
+  private def uploadAsset(content: Array[Byte] = "Your asset content here".getBytes,
+                          contentType: String = "image/png"): UploadResponse = {
+    val uploadResponse: String = `given`()
+      .basePath("")
+      .contentType(contentType)
+      .body(content)
+    .when
+      .post(s"/upload/$ACCOUNT_ID")
+    .`then`
+      .extract
+      .body
+      .asString
+
+    val blobId: String = (Json.parse(uploadResponse) \ "blobId").as[String]
+    val size: Long = (Json.parse(uploadResponse) \ "size").as[Long]
+    val assetType: String = org.apache.james.mailbox.model.ContentType.of((Json.parse(uploadResponse) \ "type").as[String])
+      .mimeType().asString()
+    UploadResponse(blobId, assetType, size)
+  }
+
+  @Test
+  def createValidRequestShouldReturnSuccessResponse(): Unit = {
+    val uploadResponse: UploadResponse = uploadAsset()
+
+    val request: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:public:assets"],
+         |  "methodCalls": [
+         |    [
+         |      "PublicAsset/set", {
+         |        "accountId": "$ACCOUNT_ID",
+         |        "create": {
+         |          "4f29": {
+         |            "blobId": "${uploadResponse.blobId}"
+         |          }
+         |        }
+         |      }, "0"
+         |    ]
+         |  ]
+         |}""".stripMargin
+
+    val response: String = `given`()
+      .body(request)
+    .when()
+      .post()
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |    "PublicAsset/set",
+           |    {
+           |        "accountId": "$ACCOUNT_ID",
+           |        "oldState": "$${json-unit.ignore}",
+           |        "newState": "$${json-unit.ignore}",
+           |        "created": {
```

Comment:
Reminder: we are not mandated to return values that are the one passed in the /set create request and that did not change...

Reply:
I propose that we should return only the following:

- `id`
- `publicURI`
- `identityIds` 

We should remove the following:

- [SHOULD] `blobId`: As you wrote, it is the "internal blobId", and the client doesn't get any benefit from it.
- [MAYBE] `size` + `contentType`: these are already known when the client uploads the content and when getting the public asset content (by response header). This will be addressed in https://github.com/linagora/tmail-backend/issues/1045

Reply:
WDYT @team ?

Reply:
No blobId, if I'm correct it's generated via hashing the payload, that should stay the same then when copying it.

No identityIds either, not server set. If you dont pass it it's null, if you pass it to create the value does not change. 

size and contentType I'm unsure but would be in favor to keep them. Technically not passed via the create request, so should return them.

Reply:
`size` and `contentType` are needed as they are infered from the blob and not explicitly passed by the client.

Reply:
=> Remove only "identityIds"